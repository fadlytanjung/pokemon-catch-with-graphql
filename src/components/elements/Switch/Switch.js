import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Switch(props) {
  const { className, inputProps, text } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');
  return (
    <input
      {...inputProps}
      className={customClass}
      data-no={text[1]}
      data-yes={text[0]}
      type="checkbox"
    />
  );
}

Switch.defaultProps = {
  className: '',
  inputProps: {},
  text: [],
};

Switch.propTypes = {
  className: PropTypes.string,
  inputProps: PropTypes.object,
  text: PropTypes.arrayOf(PropTypes.string),
};
