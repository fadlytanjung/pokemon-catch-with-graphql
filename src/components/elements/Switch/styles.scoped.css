.root {
  appearance: none;
  background-color: #ee3124;
  border-radius: 0.75rem;
  cursor: pointer;
  height: 1.5rem;
  position: relative;
  transition: background-color 0.2s ease-in-out;
  width: 3rem;
}

.root::after {
  background-color: #ffffff;
  border-radius: 50%;
  content: '';
  height: 1.25rem;
  position: absolute;
  top: 0.125rem;
  transition: left 0.2s ease-in-out;
  width: 1.25rem;
}

.root:not(:checked) {
  background-color: #ccc;
}

.root:not(:checked)::after {
  left: 0.125rem;
}

.root:checked::after {
  left: 1.625rem;
}

.root:before {
  color: #fff;
  font-size: 0.8rem;
  font-weight: 400;
  position: absolute;
  text-transform: uppercase;
  top: 0.25rem;
}

.root:not(:checked)::before {
  content: attr(data-no);
  left: 1.66rem;
}

.root:checked::before {
  content: attr(data-yes);
  right: 1.8rem;
}

@media only screen and (max-width: 48em) {
  .root:before {
    font-weight: 600;
    top: 0.2542rem;
  }
}
