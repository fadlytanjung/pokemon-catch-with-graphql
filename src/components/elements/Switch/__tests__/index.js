import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Switch from '../Switch';

describe('src/components/elements/Switch', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Switch />);
    expect(tree).toMatchSnapshot();
  });
});
