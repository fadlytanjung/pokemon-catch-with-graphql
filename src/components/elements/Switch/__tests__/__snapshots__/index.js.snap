// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Switch render 1`] = `
<input
  className=""
  type="checkbox"
/>
`;
